cnf-certification-test/accesscontrol/suite.go

```diff
@@ -54,188 +54,312 @@ var (
 	}
 )
 
-var _ = ginkgo.Describe(common.AccessControlTestKey, func() {
+var (
+	env provider.TestEnvironment
 
-	logrus.Debugf("Entering %s suite", common.AccessControlTestKey)
-	var env provider.TestEnvironment
-	ginkgo.BeforeEach(func() {
+	beforeEachFn = func(check *checksdb.Check) error {
+		logrus.Infof("Check %s: getting test environment.", check.ID)
 		env = provider.GetTestEnvironment()
-	})
+		return nil
+	}
+
+	skipIfNoContainersFn = func() (bool, string) {
+		if len(env.Containers) == 0 {
+			logrus.Warnf("No containers to check...")
+			return true, "There are no containers to check. Please check under test labels."
+		}
+
+		return false, ""
+	}
+
+	skipIfNoPodsFn = func() (bool, string) {
+		if len(env.Pods) == 0 {
+			logrus.Warn("No pods to check.")
+			return true, "There are no pods to check."
+		}
+
+		return false, ""
+	}
+
+	skipIfNoNamespacesFn = func() (bool, string) {
+		if len(env.Namespaces) == 0 {
+			warnStr := "No namespaces to check."
+			logrus.Warn(warnStr)
+			return true, warnStr
+		}
+
+		return false, ""
+	}
+
+	skipIfDaemonsetFailedToSpawnFn = func() (bool, string) {
+		warnStr := "Debug Daemonset failed to spawn skipping test."
+		if env.DaemonsetFailedToSpawn {
+			logrus.Warn(warnStr)
+			return true, warnStr
+		}
+
+		return false, ""
+	}
+
+	skipIfSharedProcessNamespacePodsFn = func() (bool, string) {
+		warnStr := "No shared process namespace pods to check."
+		if len(env.GetShareProcessNamespacePods()) == 0 {
+			logrus.Warn(warnStr)
+			return true, warnStr
+		}
+
+		return false, ""
+	}
+
+	skipIfNoCrdsFn = func() (bool, string) {
+		warnStr := "No CRDs to check."
+		if len(env.Crds) == 0 {
+			logrus.Warn(warnStr)
+			return true, warnStr
+		}
+
+		return false, ""
+	}
+
+	skipIfNoRolesFn = func() (bool, string) {
+		warnStr := "No roles to check."
+		if len(env.Roles) == 0 {
+			logrus.Warn(warnStr)
+			return true, warnStr
+		}
+
+		return false, ""
+	}
+)
+
+//nolint:funlen
+func init() {
+	logrus.Debugf("Entering %s suite", common.AccessControlTestKey)
+
+	checksGroup := checksdb.NewChecksGroup(common.ObservabilityTestKey).
```

Comment:
This change is still needed:
common..ObservabilityTestKey -> common.AccessControlTestKey.